<a href="https://colab.research.google.com/github/anabiarochar/Mestrado_AlgoritmosProgramacao/blob/main/Projeto1/3_Inser%C3%A7%C3%A3o.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [18]:
import requests
import time

# Importar dados
#url = "https://raw.githubusercontent.com/anabiarochar/Mestrado_AlgoritmosProgramacao/main/Projeto1/Vetores/vetor_100000_decrescente.txt"
url = "https://raw.githubusercontent.com/anabiarochar/Mestrado_AlgoritmosProgramacao/refs/heads/main/Projeto1/Vetores/vetor_1000_aleatorio.txt?token=GHSAT0AAAAAADXG3NI425JILOEHF3L3VXEI2NOH27Q"

try:
    response = requests.get(url)
    content = response.text

    # Ler arquivo como vetor separado por vírgula
    vetor = [int(x.strip()) for x in content.split(',') if x.strip()]

except Exception as e:
    print(f"Erro ao carregar ou processar o arquivo: {e}")

# Método Inserção
def insertionSort(A, n):
    for j in range(1, n):
        x = A[j]
        i = j - 1

        while i >= 0 and A[i] > x:
            A[i + 1] = A[i]
            i -= 1
        A[i + 1] = x
    return A

# Execução e Tempo
if vetor:

    inicio = time.perf_counter()
    vetor_para_ordenar = vetor.copy()
    vetor_ordenado = insertionSort(vetor_para_ordenar, len(vetor_para_ordenar))
    fim = time.perf_counter()

    tempo_total = fim - inicio

    print(f"Tempo de execução: {tempo_total:.6f} segundos")
    print(vetor_ordenado[:10])
else:
    print("Não foi possível processar o vetor.")

Tempo de execução: 0.022740 segundos
[1, 3, 8, 9, 9, 10, 10, 10, 10, 12]


###Aleatório 1000:
Tentativa 1 - Tempo de execução: 0.049435 segundos  
Tentativa 2 - Tempo de execução: 0.023214 segundos  
Tentativa 3 - Tempo de execução: 0.023157 segundos  
Tentativa 4 - Tempo de execução: 0.022740 segundos  

###Aleatório 10000:
Tentativa 1 - Tempo de execução: 2.556265 segundos  
Tentativa 2 - Tempo de execução: 2.521852 segundos  
Tentativa 3 - Tempo de execução: 2.684992 segundos  
Tentativa 4 - Tempo de execução: 2.498175 segundos  

###Aleatório 100000:
Tentativa 1 - Tempo de execução: 400.612380 segundos   
Tentativa 2 - Tempo de execução: 455.484334 segundos  
Tentativa 3 - Tempo de execução:  
Tentativa 4 - Tempo de execução:  

###Crescente 1000:
Tentativa 1 - Tempo de execução: 0.006639 segundos    
Tentativa 2 - Tempo de execução: 0.000185 segundos  
Tentativa 3 - Tempo de execução: 0.000141 segundos  
Tentativa 4 - Tempo de execução: 0.000107 segundos

###Crescente 10000:
Tentativa 1 - Tempo de execução: 0.000928 segundos  
Tentativa 2 - Tempo de execução: 0.001567 segundos  
Tentativa 3 - Tempo de execução: 0.000901 segundos  
Tentativa 4 - Tempo de execução: 0.000945 segundos  

###Crescente 100000:
Tentativa 1 - Tempo de execução: 0.010484 segundos  
Tentativa 2 - Tempo de execução: 0.008961 segundos  
Tentativa 3 - Tempo de execução: 0.009683 segundos  
Tentativa 4 - Tempo de execução: 0.009491 segundos  

###Decrescente 1000:
Tentativa 1 - Tempo de execução: 0.029543 segundos  
Tentativa 2 - Tempo de execução: 0.047466 segundos  
Tentativa 3 - Tempo de execução: 0.047212 segundos  
Tentativa 4 - Tempo de execução: 0.043740 segundos   

###Decrescente 10000:
Tentativa 1 - Tempo de execução: 3.203136 segundos   
Tentativa 2 - Tempo de execução: 3.738098 segundos   
Tentativa 3 - Tempo de execução: 3.283708 segundos   
Tentativa 4 - Tempo de execução: 3.225976 segundos   

###Decrescente 100000:
Tentativa 1 - Tempo de execução: 349.920139 segundos    
Tentativa 2 - Tempo de execução: 348.336984 segundos   
Tentativa 3 - Tempo de execução:  
Tentativa 4 - Tempo de execução:  